src/main/java/org/truffleruby/core/queue/QueueNodes.java

```diff
@@ -103,8 +93,30 @@ private Object doPop(UnsizedQueue queue) {
             return getContext().getThreadManager().runUntilResult(this, queue::take);
         }
 
+        @Specialization(guards = "!nonBlocking")
+        protected Object popBlocking(RubyQueue self, boolean nonBlocking, long timeoutMilliseconds,
+                @Exclusive @Cached BranchProfile closedProfile) {
+            final UnsizedQueue queue = self.queue;
+
+            var result = getContext().getThreadManager().runUntilResult(this, () -> {
+                final Object value = queue.poll(timeoutMilliseconds);
```

Comment:
This is not correct, because if it's interrupted it will try the full `timeoutMilliseconds` again, and so wait longer than the given timeout.
We should move the logic from `receive_timeout` here, or reuse it in `Queue#pop`.

There seems to be a difference in the return value, but the usages of `receive_timeout` don't seem to care about false vs nil, so let's return the same as CRuby does, and use `pop(timeout:)` instead of `receive_timeout` and remove `Queue#receive_timeout`.

Reply:
I made a bit different version. For `Queue#receive_timeout` the poll still gets the original timeout, so after an interrupt it still waits the same time instead of the remaining time. Hope that works.

Also - about removing `Queue#receive_timeout` - I would prefer doing it in a separate PR as there are some use and want this PR to be simple. Would that work?